lib/tapioca/dsl/helpers/active_record_column_type_helper.rb

```diff
@@ -235,6 +235,25 @@ def not_nilable_serialized_column?(column_type)
 
           [Array.singleton_class, Hash.singleton_class].include?(column_type.coder.object_class.singleton_class)
         end
+
+        sig { params(column_name: String).returns(T::Boolean) }
+        def nilable_column?(column_name)
+          # check db column status - if it's non-null at db, it's not nilable.
+          column = @constant.columns_hash[column_name]
+          return false if column && !column.null
+          return true unless @constant.respond_to?(:validators_on)
+
+          has_unconditional_presence_validator = @constant.validators_on(column_name).any? do |validator|
+            case validator
+            when ActiveRecord::Validations::PresenceValidator
+              !validator.options.key?(:if) && !validator.options.key?(:unless) && !validator.options.key?(:on)
+            else
+              true
```

Comment:
Shouldn't this be `false`?

Reply:
Ah yes, thank you. I changed how this was written and the conditions flipped.